src/client/Microsoft.Identity.Client/WwwAuthenticateParameters.cs

```diff
@@ -0,0 +1,307 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Identity.Client
+{
+    /// <summary>
+    /// Parameters returned by the WWW-Authenticate header. This allows for dynamic
+    /// scenarios such as claim challenge, CAE, CA auth context.
+    /// See https://aka.ms/msal-net/wwwAuthenticate.
+    /// </summary>
+    public class WwwAuthenticateParameters
+    {
+        /// <summary>
+        /// Resource for which to request scopes.
+        /// </summary>
+        public string Resource { get; set; }
+
+        /// <summary>
+        /// Scopes to request.
```

Comment:
Let's add `If it's not provided by the web API, it's computed from the Resource.`.